crowbar_framework/app/models/dns_service.rb

```diff
@@ -116,7 +116,8 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
     # If designate is enabled, we need each DNS node to be attached to the public network.
     net_svc = NetworkService.new @logger
     tnodes.each do |node|
-      if role.default_attributes[:dns][:enable_designate]
+      n = ::Node.find_by_name(node)
+      if role.default_attributes[:dns][:enable_designate] && !n.admin?
         net_svc.allocate_ip "default", "public", "host", node
       end
     end
```

Comment:
```
    # If designate is enabled, we need each DNS node to be attached to the
    # public network, except for the admin node
    net_svc = NetworkService.new @logger
    nodes = []
    tnodes.each do |node|
      n = ::Node.find_by_name(node)
      nodes << n
      if role.default_attributes[:dns][:enable_designate] && !n.admin?
        net_svc.allocate_ip "default", "public", "host", node
      end
    end
```

Reply:
Fixes the comment, and stops looping over the list twice.

Reply:
The problem here, and it's confirm after testing. Is what allocate_ip does is it takes a node name, loads it up, adds a new network IP then saves it again. However back here we have a node object loaded that is stale, doesn't have the network set in it. So when it is saved it removes what allocate_ip has just done.
When we get to the network barclamp, it's suppose to see that there is a new network and IP in the node so doesn't the actual connection magic. But because the stale version has overwritten it. It never happens.

So I like this refactor, but we either need to ignore it or:
```
   # If designate is enabled, we need each DNS node to be attached to the
    # public network, except for the admin node
    net_svc = NetworkService.new @logger
    nodes = []
    tnodes.each do |node|
      n = ::Node.find_by_name(node)
      if role.default_attributes[:dns][:enable_designate] && !n.admin?
        net_svc.allocate_ip "default", "public", "host", node
        # reload the node
        n = ::Node.find_by_name(node)
      end
     nodes << n
    end

```

Reply:
Cool the run hasn't finished yet. But I do see both nodes connected to the public network now.. so looks like my above snippet what reloads the node id allocate_ip is run works.

Reply:
pushed up Steve's refactor + the reload node addition.